conans/test/functional/build_requires/test_build_require_invalid.py

```diff
@@ -0,0 +1,317 @@
+                 "br_validate_remove_build_type.py": conanfile_validate_remove_build_type,
+                 "br_configure_remove_build_type.py": conanfile_configure_remove_build_type,
+                 "br_build_remove_build_type.py": conanfile_build_remove_build_type,
+                 "br_validate_compatible.py": conanfile_validate_compatible,
+                 "br_configure_compatible.py": conanfile_configure_compatible,
+                 "br_build_compatible.py": conanfile_build_compatible,
+                 "consumer.py": conanfile_consumer})
+    return client
+
+
+def test_create_invalid_validate(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_validate.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_invalid_configure(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_configure.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_invalid_build(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_build.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_validate_remove_build_type(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_validate_remove_build_type.py")
+    # FIXME: This should NOT fail with -> br/1.0.0: Invalid ID: br cannot be built in debug mode
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_configure_remove_build_type(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_configure_remove_build_type.py")
+    # FIXME: This should NOT fail with -> ERROR: br/1.0.0: Invalid configuration: br cannot be built in debug mode
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_build_remove_build_type(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_build_remove_build_type.py")
+    client.run("create consumer.py -s build_type=Debug")
+    assert "br/1.0.0 from local cache - Cache" in client.out
+
+
+def test_create_validate_compatible(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_validate_compatible.py")
+    client.run("create consumer.py -s build_type=Debug")
+    assert "Using compatible package" in client.out
+
+
+def test_create_configure_compatible(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_configure_compatible.py")
+    # FIXME: This should NOT fail with -> ERROR: br/1.0.0: Invalid configuration: br cannot be built in debug mode
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
```

Comment:
When using one profile, this scenario is really important too. Many checks are being raised from `configure` method, like the ones related to `cppstd` ([CMake](https://github.com/conan-io/conan-center-index/blob/master/recipes/cmake/3.x.x/conanfile.py)), which are then removed together with the `settings.compiler`.

With this error, it is not possible to use `CMake` as a build-requires and use explicit `-s compiler.cppstd=98`

Reply:
Same as above.